### PR TITLE
Reduce bundle size by lazily loading showdown

### PR DESCRIPTION
**Pull request by FrankeLukas**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).

## Motivation

See https://github.com/ls1intum/Artemis/pull/2743#pullrequestreview-576131019.

Gzip bundle size is `1.45MB` now:
grafik

## Steps for Testing

1. Make sure instructions/descriptions for any exercise types still display/render correctly (both for the instructors, and for the students in the details or exam participation views)
2. The same must work in the exam mode, too!
3. Make sure quiz questions still display and render correctly
4. Check if exercise hints, lectures, lecture units and the guided tour still display correctly
5. Student questions and answers need to work as well

**Updated:** 7. Test that the guided tour still works as expected

So tl;dr: Everything using the Markdown editor or Markdown display needs to be tested.

### CODE REVIEW

**Review by SimonEntholzer — APPROVED** [review]
The code looks fine, too :D

**Review by airelawaleria — APPROVED** [review]
could not spot any issues  :)

**Review by JohannesWeiss — APPROVED** [review]
Worked for me as well on ts1. 

One small issue was that I could not update a preexisting Lecture with a new description. Instead the following error is thrown in the console during saving: 
image
With new Lectures the issue did not occur. Since this may be unrelated as it seems to be related to the dates, approve.

**Review by marvinchimeka — APPROVED** [review]
works as intended

**Review by valeriebucher — APPROVED** [review]
I didn't encounter any issues!

**Review by sebastiantpeter — APPROVED** [review]
Tested on ts3. Seems to work fine.

File: src/test/javascript/spec/component/lecture/lecture-attachments.component.spec.ts [hunk 1]
```diff
@@ -80,7 +82,7 @@ describe('LectureAttachmentsComponent', () => {
     beforeEach(async () => {
         return TestBed.configureTestingModule({
             imports: [TranslateModule.forRoot(), ArtemisTestModule, TreeviewModule.forRoot(), RouterTestingModule.withRoutes([]), ArtemisSharedModule, FormDateTimePickerModule],
-            declarations: [LectureAttachmentsComponent],
+            declarations: [LectureAttachmentsComponent, MockPipe(HtmlForMarkdownPipe)],
```
Comment (derLalla) [inline src/test/javascript/spec/component/lecture/lecture-attachments.component.spec.ts:85]: is the MockPipe(HtmlForMarkdownPipe) still needed although you have removed the pipe in the html-template? Maybe I am wrong, but I didn't see the pipe being added somewhere. 😅

**Review by derLalla — APPROVED** [review]
I have just a small note, besides that the code has really good changes.

File: src/test/javascript/spec/component/lecture/lecture-attachments.component.spec.ts (hunk 1 above)
Comment (FrankeLukas) [reply src/test/javascript/spec/component/lecture/lecture-attachments.component.spec.ts:85]: I removed the pipe from the Guided Tour template, it is still present in the lecture-attachment one.

**Review by tiffca — APPROVED** [review]
Tested successfully

**Review by nicolasruscher — APPROVED** [review]
Nice PR, good changes! 👍

**Review by krusche — APPROVED** [review]
